### DISCUSSION TITLE
Vue 相关问题

### DISCUSSION THREAD

**Discussion by zqm971210** [zxuqian/frontend-questions#4] [求助]
## 请问up我现在,在做一个vue+django的一个项目,我在配置接口地址的时候遇到一个问题？

*在我的项目中,我用了这样的方法来配置了本地地址和线上地址*
![image](https://user-images.githubusercontent.com/48125795/96716575-cd686780-13d7-11eb-980b-0d499a48b6fe.png)

*但是我在构建的时候有的时候忘记更改这个地方,就需要在重新构建一遍.*

## 请问有有没有什么好的方法来避免这样的情况呢?

*我是刚接触前端方面的项目,对前端项目不是很熟悉想问一下如果您遇到这样的问题您会怎么解决呢?*

## 在django中我也遇到了这个问题。
![image](https://user-images.githubusercontent.com/48125795/96717141-9c3c6700-13d8-11eb-917d-0e248bbca07b.png)
我目前是这样处理的

**html**

**Comment by zxuqian** [comment]
是用的 vue cli 创建的项目吗？Vue cli 会在本地开发的时候，把 Node 的环境变量设置为 development，在打包的时候设置为 production。可以在代码里边使用 if(process.env.NODE_ENV === 'production') {} 这种判断是在哪种环境下，根据不同的环境给 API 设置不同的值，跟上边 Django 的做法类型。

你可以在 baseConfig.js 里 console.log()一下 process.env.NODE_ENV 的值。

**Comment by zqm971210** [comment]
好的 谢谢

**Comment by zxuqian** [comment]
有问题可以继续评论～

**Comment by zqm971210** [comment]
好的,谢谢,可以推荐几个 git 上的 vue 项目吗 ?想学习一下

**Comment by zxuqian** [comment]
是要学习源代码吗？我觉得在 Github 上搜索 Vue，Star 数较高的都可以用来学习。

[awesome-vue](https://github.com/vuejs/awesome-vue)

这个仓库里收集了跟 Vue 相关的资源，可以看看。

[Vuetify](https://github.com/vuetifyjs/vuetify)

Material Design 风格的 Vue 组件

**Comment by zqm971210** [comment]
好的 谢谢
